pkgs/by-name/an/antora/test/default.nix

```diff
@@ -0,0 +1,56 @@
+{
+  antora,
+  antora-lunr-extension,
+  antora-lunr-extension-test ? false,
+  antora-ui-default,
+  gitMinimal,
+  lib,
+  stdenvNoCC,
+}:
+stdenvNoCC.mkDerivation {
+  buildPhase =
+    let
+      playbook = builtins.toFile "antora-playbook.json" (
+        builtins.toJSON {
+          content.sources = [ { url = "~+"; } ];
+          runtime.log.failure_level = "warn";
+        }
+      );
+    in
+    ''
+      # The --to-dir and --ui-bundle-url options are not included in the
+      # playbook due to Antora and Nix limitations.
+      antora ${
+        lib.cli.toGNUCommandLineShell { } {
+          cache-dir = "$(mktemp --directory)";
+          extension = if antora-lunr-extension-test then antora-lunr-extension else false;
+          to-dir = placeholder "out";
+          ui-bundle-url = antora-ui-default;
+        }
+      } "${playbook}"
+    '';
+
+  postPatch = ''
+    # > In order to use a local content repository with Antora, even when using
+    # > the worktree (HEAD), the repository must have at least one commit.
+    # >
+    # > -- https://docs.antora.org/antora/3.1/playbook/content-source-url
+    git init &&
+      git \
+      -c user.email="" \
+      -c user.name="name" \
+      commit \
+      --allow-empty \
+      --allow-empty-message \
+      --message ""
```

Comment:
Check out https://github.com/NixOS/nixpkgs/blob/36dd24b460b6c4c7a64cf44bdfbe2f57ed6437ef/pkgs/development/compilers/flutter/flutter.nix#L78-L83 for a more "nixpkgs" way to do this

I don't have a big problem with this though.